### DISCUSSION TITLE
Error: installation in ubuntu (windows app)

### DISCUSSION THREAD

**Discussion by Hirshikesh** [thelfer/tfel#340] [Q&A]
I am trying to install in windows using ubuntu app, but I am getting error. Please find the attached screenshot. Is it possible to install in windows using ubuntu app. Thanks for help.
![Screenshot 2022-11-15 130602](https://user-images.githubusercontent.com/32254574/201858208-0d105269-b192-4770-9205-8ea1de56e0f3.png)

**Comment by thelfer** [comment]
Hi @Hirshikesh,

You are probably trying to install TFEL version >= 4.0 which requires a decent C++ compiler. I am quite surprise that the version of ubuntu delivered with windows to be that old...

For the sake of curiosity, would you just print the version of gcc available.

Then, you have two choices:

1. Install a more recent compiler (apt search gcc and see what is available out of the box)
2. Use TFEL version 3.4.x

With kind regards,

Thomas Helfer

**Reply by Hirshikesh** [reply]
Thank you so much for your reply. output of gcc --version is

gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0

best regards
Hirshikesh

**Reply by thelfer** [reply]
A bit too old for TFEL-4.x :)

**Comment by fpguillet** [comment]
I'm not a big fan of using MS Store for installing/running Ubuntu on WSL. I personnally use LxRunOffline (installed through Chocolatey) but WSL on the command line can also install any Ubuntu distribution available, server versions have wsl images that you can download and install manually.
Current LTS Ubuntu version is 22.04 and is also available through MS Store.  It's a good idea to keep your WSL ubuntu  up to date using apt dist-upgrade before you're stuck with outdated packages.
